controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -447,60 +447,65 @@ func (r *DataScienceClusterReconciler) SetupWithManager(mgr ctrl.Manager) error
 			builder.WithPredicates(argoWorkflowCRDPredicates)).
-		Watches(&source.Kind{Type: &corev1.Secret{}}, handler.EnqueueRequestsFromMapFunc(r.watchDefaultIngressSecret), builder.WithPredicates(defaultIngressCertSecretPredicates)).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, handler.EnqueueRequestsFromMapFunc(r.watchDefaultIngressSecret(ctx)), builder.WithPredicates(defaultIngressCertSecretPredicates)).
 		// this predicates prevents meaningless reconciliations from being triggered
 		WithEventFilter(predicate.Or(predicate.GenerationChangedPredicate{}, predicate.LabelChangedPredicate{})).
 		Complete(r)
 }
 
-func (r *DataScienceClusterReconciler) watchDataScienceClusterForDSCI(a client.Object) []reconcile.Request {
-	requestName, err := r.getRequestName()
-	if err != nil {
+func (r *DataScienceClusterReconciler) watchDataScienceClusterForDSCI(ctx context.Context) func(client.Object) []reconcile.Request {
+	return func(a client.Object) []reconcile.Request {
+		requestName, err := r.getRequestName(ctx)
+		if err != nil {
+			return nil
+		}
+		// When DSCI CR gets created, trigger reconcile function
+		if a.GetObjectKind().GroupVersionKind().Kind == "DSCInitialization" || a.GetName() == "default-dsci" {
+			return []reconcile.Request{{
+				NamespacedName: types.NamespacedName{Name: requestName},
+			}}
+		}
 		return nil
 	}
-	// When DSCI CR gets created, trigger reconcile function
-	if a.GetObjectKind().GroupVersionKind().Kind == "DSCInitialization" || a.GetName() == "default-dsci" {
-		return []reconcile.Request{{
-			NamespacedName: types.NamespacedName{Name: requestName},
-		}}
-	}
-	return nil
 }
-func (r *DataScienceClusterReconciler) watchDataScienceClusterResources(a client.Object) []reconcile.Request {
-	requestName, err := r.getRequestName()
-	if err != nil {
-		return nil
-	}
 
-	if a.GetObjectKind().GroupVersionKind().Kind == "CustomResourceDefinition" || a.GetName() == "ArgoWorkflowCRD" {
-		return []reconcile.Request{{
-			NamespacedName: types.NamespacedName{Name: requestName},
-		}}
-	}
+func (r *DataScienceClusterReconciler) watchDataScienceClusterResources(ctx context.Context) func(client.Object) []reconcile.Request {
+	return func(a client.Object) []reconcile.Request {
+		requestName, err := r.getRequestName(ctx)
+		if err != nil {
+			return nil
+		}
 
-	// Trigger reconcile function when uninstall configmap is created
-	operatorNs, err := cluster.GetOperatorNamespace()
-	if err != nil {
-		return nil
-	}
-	if a.GetNamespace() == operatorNs {
-		cmLabels := a.GetLabels()
-		if val, ok := cmLabels[upgrade.DeleteConfigMapLabel]; ok && val == "true" {
+		if a.GetObjectKind().GroupVersionKind().Kind == "CustomResourceDefinition" || a.GetName() == "ArgoWorkflowCRD" {
 			return []reconcile.Request{{
 				NamespacedName: types.NamespacedName{Name: requestName},
 			}}
 		}
+
+		// Trigger reconcile function when uninstall configmap is created
+		operatorNs, err := cluster.GetOperatorNamespace()
```

Comment:
not sure if we want to pass in ctx in GetOperatorNamespace() as well? does not seem needed

Reply:
The function is defined as:

```go
func GetOperatorNamespace() (string, error) {
	data, err := os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
	return string(data), err
}
```

So it does not require a `Context`